wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -557,11 +556,12 @@ def handle_generic_atomic(emitter: WaveEmitter, node: fx.Node):
 
 
 def get_rank(mlir_type):
-    if not isinstance(mlir_type, ShapedType):
+    if hasattr(mlir_type, "rank"):
```

Comment:
This is another questionable change.  The vector types that I'm generating are not caught here.  Maybe I should find another type to test.  But this was also a mildly frustrating problem to track down to having a whitelisted set of values that didn't catch mine, so honestly I think I prefer the duck-type version.